homeassistant/components/zha/light.py

```diff
@@ -331,8 +386,9 @@ async def async_turn_off(self, **kwargs):
             return
         self._state = False
 
-        if duration and supports_level:
+        if supports_level:
             # store current brightness so that the next turn_on uses it.
+            self._off_with_transition = bool(duration)
             self._off_brightness = self._brightness
```

Comment:
Another thing I thought of:

When the light is turned off while it's already off (using a service call for example), the ``off_brightness`` gets overridden (and set to 1 apparently).
This makes the next turn on very dim (in cases where ``off_brightness`` needs to be used).

In current releases that only happens if ``turn_off`` is called with some kind of ``transition``

There should probably be a guard to only set the ``off_brightness`` if the light is currently on.